odoo/tests/common.py

```diff
@@ -224,6 +225,66 @@ def assertQueryCount(self, default=0, margin=0, **counters):
         else:
             yield
 
+    def assertRecordsEqual(self, records, expected_values):
+        ''' Compare a recordset with a list of dictionaries representing the expected results.
+        This method performs a comparison element by element based on their index.
+        Then, the order of the expected values is extremely important.
+
+        Note that:
+          - Comparison between falsy values is supported: False match with None.
+          - Comparison between monetary field is also treated according the currency's rounding.
+          - Comparison between x2many field is done by ids. Then, empty expected ids must be [].
+          - Comparison between many2one field id done by id. Empty comparison can be done using any falsy value.
+
+        :param records:               The records to compare.
+        :param expected_values:       List of dicts expected to be exactly matched in records
+        '''
+
+        def _compare_candidate(record, candidate):
+            ''' Return True if the candidate matches the given record '''
+            for field_name in candidate.keys():
+                record_value = record[field_name]
+                candidate_value = candidate[field_name]
+                field_type = record._fields[field_name].type
+                if field_type == 'monetary':
+                    # Compare monetary field.
+                    currency_field_name = record._fields[field_name]._related_currency_field
+                    record_currency = record[currency_field_name]
+                    if record_currency.compare_amounts(candidate_value, record_value)\
+                            if record_currency else candidate_value != record_value:
+                        return False
+                elif field_type in ('one2many', 'many2many'):
+                    # Compare x2many relational fields.
+                    # Empty comparison must be an empty list to be True.
+                    if not record_value.ids == candidate_value:
+                        return False
+                elif field_type == 'many2one':
+                    # Compare many2one relational fields.
+                    # Every falsy value is allowed to compare with an empty record.
+                    if (record_value or candidate_value) and record_value.id != candidate_value:
+                        return False
+                elif (candidate_value or record_value) and record_value != candidate_value:
+                    # Compare others fields if not both interpreted as falsy values.
+                    return False
+            return True
+
+        def _format_message(records, expected_values):
+            ''' Return a formatted representation of records/expected_values. '''
+            all_records_values = records.read(expected_values[0].keys(), load=False)
```

Comment:
It's nicer to cast dict views such as dict.keys() into lists when they leave your scope, just to play nice with other people's expectations.